### ISSUE TITLE
Add `FormBuilderLoader` in-line on blocks

### ISSUE THREAD

**Issue by ScopeyNZ** [dnadesign/silverstripe-elemental#327]
The designs show a collapsible section where a form will appear when expanded:

https://projects.invisionapp.com/share/ZRL4T2HQAGB#/screens/301923214

![image](https://user-images.githubusercontent.com/5170590/44313723-8847f400-a461-11e8-9606-5b315e14ef76.png)

Additionally a custom React form component will have to be used as we can't use form tags - we're rendering within the top-level page form

A/Cs:
- [x] ~Implement a collapsible section per the designs~
- [x] On expand - load a form with `FormBuilderLoader` using the schema endpoint (implemented by #326)
- [x] ~Expanding and collapsing the block is keyboard accessible.~
- [x] The form is keyboard accessible
- [x] Form is not removed when collapsed
- [x] Form is not reloaded on subsequent expand
- [x] A custom form react component is used (by setting the `SchemaComponent` on a form) to remove the use of a `<form>` tag

Split from #295 
Expanding functionality has been tasked out https://github.com/silverstripe/silverstripe-elemental-blocks/issues/43

## Pull requests

- [x] Add schema information for HTMLEditorField silverstripe/silverstripe-framework#8359
- [x] NEW HtmlEditorField component for react rich text silverstripe/silverstripe-admin#628
- [x] Add ability to load Element's edit form inline #372

**Comment by brynwhyman** [comment]
This looks like it could replace the placeholder issue we created for adding expand functionality: https://github.com/silverstripe/silverstripe-elemental-blocks/issues/43 

If so, I'll update the ACs here

**Comment by ScopeyNZ** [comment]
Ah yes. I was looking for that. It's on a different repo which could explain why I wasn't able to find it.

**Comment by newleeland** [comment]
**Additional design reference (Style guide)**
https://invis.io/7ZNLP773RSY

**Complete blocks style guide**
Blocks: https://invis.io/T7NLP7WP3K4
Layout blocks: https://invis.io/BNNLP87VQJU

**Comment by raissanorth** [comment]
The Behat tests will need to have a scenario added that checks if the element is editable. This has been removed in https://github.com/dnadesign/silverstripe-elemental/pull/353, as at that point in time only a placeholder was rendered.

**Comment by robbieaverill** [comment]
I’m adding a class when the element is _not_ editable at the moment as part of another ticket. You think that’ll be ok? I’ll update the behat test to cover the dropdown menu being there or not. Maybe will add userforms to the tests as a concrete element which doesn’t allow it

**Comment by NightJar** [comment]
Required to resolve this issue:
- [ ] Add schema information for HTMLEditorField silverstripe/silverstripe-framework#8359
- [ ] NEW HtmlEditorField component for react rich text silverstripe/silverstripe-admin#628
- [ ] Add ability to load Element's edit form inline #372

**Comment by robbieaverill** [comment]
I've just given it a test run - some comments:

* The TinyMCE editor only initialised when I open the console originally, now it's loading immediately (perhaps a timing/browser caching issue of the resources for it)
* The editor is too big - perhaps restrict the size of it to around 150-200px? You can do this with `->setRows(n)`
* The form fields have divider lines under each form field. You can fix this by adding the `form--no-dividers` class to the form
* It's pretty minor, but I think I'd expect the loading indicator for a block when you trigger the inline form to open to be shown on the block itself, not the whole block area
* Side issue: the history viewer still isn't working in the React form context - I'd hoped that https://github.com/silverstripe/silverstripe-versioned-admin/pull/55 would have resolved that but I guess it was wishful thinking. I've logged this separately as https://github.com/silverstripe/silverstripe-versioned-admin/issues/59

**Comment by NightJar** [comment]
Yes there is some minor issue with TinyMCE. Simply clicking the area under the label will trigger the render as well (weirdly, the same is _not_ true if one clicks where TinyMCE should be). It's a rather bizarre.

I really don't know how to get around this without spending a whole lot of time porting the entire thing (i.e. our plugins) to react - as per https://github.com/silverstripe/silverstripe-admin/issues/623

**Comment by NightJar** [comment]
OK, I figured out what triggered the initial render - it must be an _Entwine_ event.
E.g. clicking where the editor should have been is a native event, hovering the site tree activates an entwine hover event & causes the render.

https://github.com/hafriedlander/jquery.entwine/blob/master/src/domevents/jquery.entwine.domevents.addrem.js#L132-L134

Luckily I was able to track down a way to manually trigger entwine into updating, and have updated the admin PR with such 💥
